### PR TITLE
Added CLI for M365 version of script "SharePoint Modern Page Publishing Report"

### LINKED ISSUE #305: CLI for M365 version for script "SharePoint Modern Page Publishing Report"
🎯Aim
The aim of this issue would be to add a [CLI for M365](https://pnp.github.io/cli-microsoft365/) version of script [SharePoint Modern Page Publishing Report](https://pnp.github.io/script-samples/spo-modern-page-publishing-report/README.html?tabs=pnpps).

🤔How
I think one of the best approach would be to use the following commands:

- [listitem-list](https://pnp.github.io/cli-microsoft365/cmd/spo/listitem/listitem-list/)
- or [page-list](https://pnp.github.io/cli-microsoft365/cmd/spo/page/page-list/)
- or mix of both 😉

### PR DESCRIPTION
**Pull request by nanddeepn**
Added CLI for M365 version of script [SharePoint Modern Page Publishing Report](https://pnp.github.io/script-samples/spo-modern-page-publishing-report/README.html?tabs=pnpps).
Closes #305

### CODE REVIEW

**Comment by nanddeepn** [comment]
Hi @Adam-it 
Please review.

Found that [spo page list](https://pnp.github.io/cli-microsoft365/cmd/spo/page/page-list/) and [spo page get](https://pnp.github.io/cli-microsoft365/cmd/spo/page/page-get/) does not seem to return the `_ModerationStatus`
Can you please confirm?

**Comment by Adam-it** [comment]
yes that seems to be the case for default settings for a pages lib
image

in order for this metadata to be returned by the SP rest API we need to have the approval flow settings turned on in the site pages library
image

after the above is set and saved we may get the info from `.ListItemAllFields.OData__ModerationStatus` property
image


Hope it helped 😉

File: scripts/spo-modern-page-publishing-report/README.md [hunk 1]
```diff
@@ -123,11 +123,87 @@ process{
...
+        Write-Host " Processing page $($modernPage.ServerRelativeUrl)" -ForegroundColor Cyan
+
+        switch ($modernPage.ListItemAllFields.PromotedState) {
+            0 { $promotedLevel = "Page" }
+            1 { $promotedLevel = "News" }
+            2 { $promotedLevel = "News" }
+            Default {}
+        }
+
+        switch ($modernPage.ListItemAllFields._ModerationStatus) {
```
Comment (Adam-it) [inline scripts/spo-modern-page-publishing-report/README.md:169]: should be `OData__ModerationStatus`

**Review by Adam-it — COMMENTED** [review]
pulled locally and tested. Works ✅
you rock 🤩

**Pull request merged**